TST: Add regression test for Series dropping uint datatype (#18311)

gorski-m opened the pull request:
- [x] closes #18311
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Commit:
TST: Add regression test for Series dropping uint datatype (#18311)

WillAyd commented on pandas/tests/series/test_dtypes.py:
```diff
@@ -518,3 +518,9 @@ def test_values_compatibility(self, data):
         tm.assert_numpy_array_equal(result, expected)
+
+    def test_uint_drop(self):
+        for type in ["uint8", "uint16", "uint32", "uint64"]:
```
Could you:

  1. Parameterize this instead of doing a loop within the function body
  2. Change `type` to `type_` so as not to shadow the builtin
  3. Add the issue number as a comment?

jreback replied:
you can pop in ``uint_dtype`` to the argument and it will just work

WillAyd reviewed (changes requested):
Thanks for the PR. Some minor edits

jreback commented on pandas/tests/series/test_dtypes.py:
```diff
@@ -518,3 +518,9 @@ def test_values_compatibility(self, data):
+        for type in ["uint8", "uint16", "uint32", "uint64"]:
+            series = pd.Series([1, 2, 3], dtype=type)
+            series.loc[0] = 4
+            assert series.dtype == type
```
create the expected series and user tm.assert_series_equal(series, expected)

jreback commented on pandas/tests/series/test_dtypes.py:
```diff
@@ -518,3 +518,9 @@ def test_values_compatibility(self, data):
         expected = np.array(data.astype(object))
         tm.assert_numpy_array_equal(result, expected)
+
+    def test_uint_drop(self):
```
move to pandas/series/test_indexing.py

Commit:
Move test to test_indexing.py and resolve requested changes

Requested changes were:
- add issue number as a comment
- parametrize test, instead of looping through array
- change type variable to type_

jreback commented on pandas/tests/series/indexing/test_indexing.py:
```diff
@@ -907,3 +907,13 @@ def test_head_tail(test_data):
     assert_series_equal(test_data.series.tail(0), test_data.series[0:0])
+
+
+@pytest.mark.parametrize("type_", ["uint8", "uint16", "uint32", "uint64"])
```
can you use the uint_dtype fixture (literally drop the parametrize), and pass uint_dtype as the arg (and rename type_ -> uint_dtype), ping on green.

jreback replied:
actually might as well use 'any_int_dtype' here

Commit:
Change function argument to any_int_type and dropped parametrize

WillAyd reviewed (approved):
lgtm @jreback

jreback commented:
thanks @Reksbril
